### PR TITLE
Allow more than one person to run, generalize, modularize, code.

### LINKED ISSUE #1: Allow second person to run
Get `mk_lc.sh`, `mk_lc.py` generalized and lighty refactor to allow a second person, in this case wmwv, to run them.

1. [x] The basic conceptual unit of work is a SN.  The basic unit of work is a SN+band.  Structure running around this
2. [x] Store environment variables, config variables for any filepaths.
3. [x] If it's in a for loop it should go into a function
4. [x] Wrap in argparse.

### PR DESCRIPTION
**Pull request by wmwv**
Generalizes input, output dirs.  I can run a lightcurve through now.

This runs from the command line.  I still need to test if the SLURM array job stuff works, but it's ready to have someone else take a look at it.

Closes #1

### CODE REVIEW

File: mk_lc.py [hunk 1]
```diff
@@ -136,241 +178,266 @@ def lc(oid, band):
...
-
-# truth['mjd'] = [row['mjd'].replace('(','') for row in truth]
-# truth['mag'] = [row['mag'].replace(')','') for row in truth]
-
-# truth['mjd'] = truth['mjd'].astype(float)
-# truth['mag'] = truth['mag'].astype(float)
-# truth['filter'] = [filt.upper() for filt in truth['filter']]
-
-# Make plot:
-# gs = GridSpec(nrows=6,ncols=3, height_ratios=[1,.33,1,.33,1,.33],width_ratios=[1,1,1],wspace=0,hspace=0)
```
Comment (laldoroty) [inline mk_lc.py:303]: I definitely want to keep this function--it makes a plot with all LCs for one SN after all bands have run. I guess there should be a different place for this, maybe in phrosty.plotting?

**Review by laldoroty — CHANGES_REQUESTED** [review]
Only comment is on the function that makes the plot with all LCs for a single SN.

**Comment by wmwv** [comment]
Did the latest commit capture the function you wanted?

I don't expect it to run, e.g., the function only takes `oid` as an argument and problem needs more, but the commands are at least preserved.

**Comment by laldoroty** [comment]
Yep!

**Review by laldoroty — APPROVED** [review]

**Pull request merged**